workflows/taxprofiler.nf

```diff
@@ -17,7 +17,26 @@ def checkPathParamList = [ params.input, params.databases, params.hostremoval_re
 for (param in checkPathParamList) { if (param) { file(param, checkIfExists: true) } }
 
 // Check mandatory parameters
-if (params.input    ) { ch_input     = file(params.input)     } else { exit 1, 'Input samplesheet not specified!' }
+if (params.input) {
+    ch_input = file(params.input)
+    ch_input_basedir = []
+
+} else if (params.pep) {
+
+    if ( params.pep.startsWith("http://") || params.pep.startsWith("https://") ) {
+        ch_input = file(params.pep)
+        ch_input_basedir = []
+    }
+
+    else {
+        ch_input = file(params.pep)
+        ch_input_basedir = new File(params.pep).getParent()
```

Comment:
I still don't fully understand what this is used for, but you can probably use the 'native' Nextflow file attributes to get the directory if necessary: https://www.nextflow.io/docs/latest/script.html#check-file-attributes